src/sdam/state/mod.rs

```diff
@@ -87,26 +87,36 @@ impl Topology {
 
         let http_client = HttpClient::default();
 
-        let servers = hosts
-            .into_iter()
-            .map(|address| {
-                (
-                    address.clone(),
-                    Server::new(address, &ClientOptions::default(), http_client.clone()).into(),
-                )
-            })
-            .collect();
-
         let state = TopologyState {
             description,
-            servers,
-            http_client,
+            servers: Default::default(),
+            http_client: http_client.clone(),
         };
 
-        Self {
+        let topology = Self {
             state: Arc::new(RwLock::new(state)),
             common,
-        }
+        };
+
+        // we can block in place here because we're the only ones with access to the lock, so it
+        // should be acquired immediately.
+        let mut topology_state = RUNTIME.block_in_place(topology.state.write());
+
+        hosts.into_iter().for_each(|address| {
```

Comment:
Out of curiosity, why not use `for address in hosts` instead?

Reply:
This used to `collect` into something so it was the fewest character change to just make it a `for_each` I think. Anyways rewritten to `for in`.